src/Identity/Core/src/Passkeys/BufferSourceJsonConverter.cs

```diff
@@ -0,0 +1,99 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Buffers;
+using System.Buffers.Text;
+using System.Diagnostics;
+using System.Diagnostics.CodeAnalysis;
+using System.Text.Json;
+using System.Text.Json.Serialization;
+
+namespace Microsoft.AspNetCore.Identity;
+
+internal sealed class BufferSourceJsonConverter : JsonConverter<BufferSource>
```

Comment:
Is this needed because S.T.J doesn't support ReadOnlyMemory<byte>?

Reply:
This would be a reasonable thing to bring to S.T.J. The reason I brought up in the first place is that this is non-trivial logic that we are copying/putting here, when this could ideally be a built-in `JsonConverter` that got applied.